File: erlang/fizzbuzz.erl
-module(fizzbuzz).
-export([fizzbuzzer/1]).

fizzbuzzer(N) ->
  if
    N rem 3 == 0 ->
      io:format("Fizz", []);
    N rem 5 == 0 ->
      io:format("Buzz", []);
    true ->
      io:format("~w", [N])
  end,
  io:format("~n"),
  if
    N > 1 ->
      fizzbuzzer(N-1);
    true ->
      io:format("Done!", [])
  end.

File: haskell/fizzbuzz.hs
module Main where

main :: IO()
main = mapM_ fizzbuzz [1..100]

fizzbuzz :: Int -> IO()
fizzbuzz n = do
  putStrLn (fizzbuzzStr n)

fizzbuzzStr :: Int -> String
fizzbuzzStr n =
  if evenDiv n 3 && evenDiv n 5
    then "FizzBuzz"
    else if evenDiv n 3 
      then "Fizz"
      else if evenDiv n 5
        then "Buzz"
        else show n

evenDiv :: Int -> Int -> Bool
evenDiv x y = (x `rem` y) == 0




